gems/aws-sdk-core/lib/aws-sdk-core/profile/profile_file_parser.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+module AWS::SDK::Core
+  # @api private
+  class ProfileFileParser
+    def initialize(profile_file_contents)
+      @profile_file_contents = profile_file_contents
+      @current_line_number = 0
+    end
+
+    def parse
+      initialize_profiles
+      @profiles
+    end
+
+    private
+
+    def initialize_profiles
+      @profiles = {}
+
```

Comment:
Nit: Since we're resetting profiles here, we should probably set @current_line_number = 0 here as well?